src/main/resources/react4xp/_entries/Dashboard/DashboardDataQuery.jsx

```diff
@@ -1,75 +1,117 @@
 import Button from 'react-bootstrap/Button'
-import { Download, Trash } from 'react-feather'
+import { RefreshCw, Trash } from 'react-feather'
 import { Link } from '@statisticsnorway/ssb-component-library'
 import React from 'react'
 import PropTypes from 'prop-types'
+import {DataQuery} from './Dashboard';
 
 class DashboardDataQuery extends React.Component {
   constructor(props) {
     super(props)
+    this.state = {
+      deleting: false
+    }
+  }
+
+  getDataset(dataQueryId) {
+    this.setLoading(true)
+    this.resultHandler(this.props.getRequest(dataQueryId))
+  }
+
+  deleteDataset(dataQueryId) {
+    this.setState({
+      deleting: true
+    })
+    return this.resultHandler(this.props.deleteRequest(dataQueryId))
+  }
+
+  resultHandler(p) {
+    return p.then((response) => {
+      if (response.status === 200) {
+        this.props.refreshRow(response.data.updates)
+        this.props.showSuccess(response.data.message)
+      } else {
+        this.props.showError(response.data.message)
+      }
+    })
+      .catch((e) => {
+        console.log(e)
+        this.props.refreshRow(e.response.data.updates)
+        this.props.showError(e.response.data.message)
+      })
+      .finally(() => {
+        this.setState({
+          deleting: false
+        })
+        this.setLoading(false)
+      })
+  }
+
+  setLoading(value) {
+    this.props.setLoading(this.props.dataquery.id, value)
+  }
+
+  renderLogData() {
+    const logData = this.props.dataquery.logData
+    return (
+      <td>
+        { logData.message ? logData.message : '' }<br/>
+        { logData.modifiedReadable ? logData.modifiedReadable : '' }<br/>
+        { logData.modified ? logData.modified : '' }<br/>
+        { logData.by.displayName ? `av ${logData.by.displayName}` : '' }
+      </td>
+    )
   }
 
   render() {
+    const dataQuery = this.props.dataquery
     return (
       <tr className="small" >
-        <td className={`${this.props.hasData ? 'ok' : 'error'} dataset`}>
-          {this.props.displayName ? <Link href={'/admin/tool/com.enonic.app.contentstudio/main#/edit/' + this.props.id}>{this.props.displayName}</Link> : ''}
-          <span className={'float-right detail ' + this.props.format}>{this.props.format}</span>
-          {!this.props.isPublished ? <span className={'float-right detail unpublished'}>Ikke publisert</span> : ''}
-          {this.props.errorMsg ? <span className="errorMsg">{this.props.errorMsg}</span> : ''}
-        </td>
 
-        <td>
-          {this.props.datasetModifiedReadable ? this.props.datasetModifiedReadable : ''}<br/>
-          {this.props.datasetModified ? this.props.datasetModified : ''}
+        <td className={`${dataQuery.hasData ? 'ok' : 'error'} dataset`}>
+          {dataQuery.displayName ?
+            <Link href={'/admin/tool/com.enonic.app.contentstudio/main#/edit/' + dataQuery.id}>
```

Comment:
https://developer.enonic.com/docs/xp/stable/api/lib-admin#gettoolurl
Should maybe use this to generate the correct base url, in case it's not hosted on /admin

Reply:
Done in separate PR 👍